aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,398 @@
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  /**
+   * All-arg constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param warehousePath warehouse path, such as s3://my-iceberg-bucket/metadata_prod
+   * @param skipArchive if Glue should skip an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to turn this feature off.
+   */
+  public GlueCatalog(
+      Configuration conf,
+      GlueClient glue,
+      String catalogId,
+      String warehousePath,
+      boolean skipArchive) {
+    this.conf = Preconditions.checkNotNull(conf, "Cannot initialize GlueCatalog because Hadoop config is null");
+    this.glue = Preconditions.checkNotNull(glue, "Cannot initialize GlueCatalog because GlueClient is null");
+    this.catalogId = catalogId;
+    this.catalogName = "glue." + catalogId; // for logging purpose
+    this.warehousePath = warehousePath;
+    this.skipArchive = skipArchive;
+    validateWarehousePath();
+  }
+
+  private void validateWarehousePath() {
+    Preconditions.checkArgument(warehousePath != null && warehousePath.length() > 0,
+        "Cannot initialize GlueCatalog because warehousePath must not be null");
+    Preconditions.checkArgument(warehousePath.charAt(warehousePath.length() - 1) != '/',
+        "Cannot initialize GlueCatalog because warehousePath %s must not end with /", warehousePath);
+  }
+
+  @Override
+  public void close() throws IOException {
+    glue.close();
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    return new GlueTableOperations(conf, glue, catalogId, skipArchive, tableIdentifier);
+  }
+
+  /**
+   * This method produces the same result as using a HiveCatalog.
+   * If databaseUri exists for the Glue database URI, the default location is databaseUri/tableName.
+   * If not, the default location is warehousePath/databaseName.db/tableName
+   * @param tableIdentifier table id
+   * @return default warehouse path
+   */
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    // check if value is set in database
+    GetDatabaseResponse response = glue.getDatabase(GetDatabaseRequest.builder()
+        .name(IcebergToGlueConverter.getDatabaseName(tableIdentifier))
+        .build());
+    String dbLocationUri = response.database().locationUri();
+    if (dbLocationUri != null) {
+      return String.format("%s/%s", dbLocationUri, tableIdentifier.name());
+    }
+    return String.format(
```

Comment:
Nit: we like to use an empty line after control flow statements to space out blocks a bit more.

Reply:
Looks like this is needed in a lot of places.

Reply:
sure, will update